packages/plugins/graphql-plugin-auth/src/Neo4jGraphQLAuthJWKSPlugin.ts

```diff
@@ -122,9 +129,15 @@ class Neo4jGraphQLAuthJWKSPlugin {
                 (err, decoded) => {
                     if (err) {
                         reject(err);
-                    } else {
-                        resolve(decoded as unknown as T);
+                        return;
+                    }
+                    if (this.issuer && decoded && typeof decoded === "object") {
```

Comment:
Actually, the issuer can be verified as part of the `jsonwebtoken.verify()` function (line 123). See [ref](https://github.com/auth0/node-jsonwebtoken#jwtverifytoken-secretorpublickey-options-callback).
I can't comment on that line 123 but it would be something like:
```
jsonwebtoken.verify(
      token,
      getKey,
      {
          algorithms: ["HS256", "RS256"],
          issuer: this.issuer,    // <-- this here
      },
      (err, decoded) => {
             if (err) {
                        reject(err);
              } else {
                        resolve(decoded as unknown as T);
              }
      }
)
```